<launch>
    <arg name="simulation" default="false" />

    <include file="$(find turtlebot_bringup)/launch/minimal.launch" unless="$(arg simulation)" />
    <include file="$(find competition3)/launch/turtlebot_world.launch" if="$(arg simulation)">
        <arg name="initial_robot_pose" value="-x -2.35 -y 3.45 -z -0.05 -R 0 -P 0 -Y -1.62" />
    </include>

    <include file="$(find competition3)/launch/ar_track.launch">
        <arg name="simulation" value="$(arg simulation)"/>
    </include>

  <rosparam command="load" file="$(find competition3)/param/simulation/named_poses.yaml" if="$(arg simulation)"/>
  <rosparam command="load" file="$(find competition3)/param/realworld/named_poses.yaml" unless="$(arg simulation)"/>
   <include file='$(find competition3)/launch/rtabmap_demo_turtlebot_mapping.launch'>
        <arg name='localization' value='true' />
        <arg name='database_path' value='$(find competition3)/rtabmap/realworld/rtabmap.db' unless="$(arg simulation)" />
        <arg name='database_path' value='$(find competition3)/rtabmap/simulation/rtabmap.db' if="$(arg simulation)" />
        <arg name="simulation" value="$(arg simulation)"/>
    </include>

    <node pkg='rviz' name='rviz' type='rviz' args="-d $(find competition3)/rviz/competition3.rviz"/>

    <!-- Convert odometry to Pose2D -->
    <node pkg="kobuki_node" type="getOdom2D.py" name="getOdom2D">
        <remap from="mobile_base/odom" to="mobile_base/odom" />
        <remap from="pose2d" to="/pose2d" />
    </node>

    <!-- Service to convert between camera pixels and real-world points -->
    <node pkg="line_follower" type="cam_pixel_to_point.py" name="cam_pixel_to_point" output="screen">
        <remap from="camera_depth_image" to="/camera/depth/image_raw" if="$(arg simulation)" />
        <remap from="camera_depth_image" to="/camera/depth_registered/image_raw" unless="$(arg simulation)" />
        <remap from="camera_info" to="/camera/rgb/camera_info" />
        <remap from="cam_pixel_to_point" to="cam_pixel_to_point" />
    </node>

    <node pkg="competition3" type="competition3.py" name="competition3" output="screen" />

    <node pkg="usb_cam" type="usb_cam_node" name="usb_cam_node" unless="$(arg simulation)" >
        <param name="video_device" value="/dev/video2"/>
        <param name="pixel_format" value="yuyv" />
        <remap from="usb_cam_node/image_raw" to="/bottom_camera/image_raw" />
    </node>

    <node pkg="joy" type="joy_node" name="joy_node"/>
</launch>
